### DISCUSSION TITLE
Trouble with 'build-finished' event

### DISCUSSION THREAD

**Discussion by rsokl** [felix-hilden/sphinx-codeautolink#52] [Q&A] [answered]
Hello! I am hitting an issue with this extension after the sphinx build succeeds. I figure it must have something to do with the structure of my docs.

[Here](https://github.com/rsokl/stubby) is a minimal project for reproducing the issue.

After running the following

```shell
stubby/docs > python -m sphinx source build
```

I get

```shell
...

dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in build.

Extension error (sphinx_codeautolink.extension):
Handler <bound method SphinxCodeAutoLink.apply_links of <sphinx_codeautolink.extension.SphinxCodeAutoLink object at 0x000001F935DDBE20>> for event 'build-finished' threw an exception (exception: [Errno 2] No such file or directory: 'C:\\Users\\Ryan Soklaski\\stubby\\docs\\build\\generated\\stubby.html')
```

Without the `sphinx-codeautolink` extension, everything builds without issue. Could you provide me with some guidance as to what I might be missing?

**Comment by felix-hilden** [comment, marked as answer]
Hmm, thanks for letting me know. For some reason it is trying to locate the wrong file. I imagine the HTML file is named `stubby.func.html`, right? With a quick test I wasn't able to reproduce by just renaming a file of mine to have a double dot, but I'll have a look at the example you provided at the end of the week!

**Reply by felix-hilden** [reply]
@rsokl I tracked the problem down: the `.func` is replaced by `pathlib.Path.with_suffix`. I don't know why my test didn't reproduce the error, but I cloned your example, made the change and it succeeds. Given that our tests also pass with it, we'll go for it. Now fixed in the latest commit!

**Reply by rsokl** [reply]
Great! I will give it a whirl and will keep you posted.

**Reply by felix-hilden** [reply]
I'll mark this as answered now, but I'd still be interested to hear from you!

**Reply by rsokl** [reply]
Ah yes, sorry for dropping of the map with this. I hope to circle back and give this another shot soon!

**Reply by rsokl** [reply]
Heya! Just FYI, I just added `0.9.0` of your extensions to my project, and it went smoothly! 

https://mit-ll-responsible-ai.github.io/hydra-zen/index.html

One thing that I noticed, is that adding the extension broke my IDE's (VSCode with [this rst extension](https://marketplace.visualstudio.com/items?itemName=lextudio.restructuredtext)) ability to preview my docs.
The following error is reported when the preview fails to generate:

```
Extension error (sphinx_codeautolink.extension):
Handler > for event 'env-updated' threw an exception (exception: 'utf-8' codec can't decode byte 0xd9 in position 4: invalid continuation byte)
```

Interestingly, I see no errors nor warning when I build the docs myself. Disabling the autolink extension fixes my IDE's ability to generate previews.

**Reply by felix-hilden** [reply]
Glad to hear! But that's odd 😄 if you'd like to dig deeper, perhaps that's a just a file encoding issue we can fix, but it's hard to guess what it could be. I don't use VSC my self.
